TESTS/icb_arpack_cpp.cpp

```diff
@@ -26,50 +25,48 @@ void diagonal_matrix_vector_product(Real const* const x, Real* const y) {
 }
 
 template <typename Real>
-void real_symmetric_runner(double const& tol_check) {
+void real_symmetric_runner(double const& tol_check, arpack::which const& ritz_option) {
   a_int const N = 1000;
   a_int const nev = 9;
 
   a_int const ncv = 2 * nev + 1;
   a_int const ldv = N;
 
-  a_int const ldz = N + 1;
+  a_int const ldz = N;
 
-  a_int const lworkl = 3 * (ncv * ncv) + 6 * ncv;
+  a_int const lworkl = ncv * (ncv + 8);
 
   Real const tol = 0.000001; // small tol => more stable checks after EV computation.
-  Real const sigma = 0.0;
+  Real const sigma = 0.0;    // not referenced in this mode
 
-  a_int const rvec = 1;
+  a_int const rvec = 1;      // need eigenvectors
 
   std::vector<Real> resid(N);
   std::vector<Real> V(ncv * N);
   std::vector<Real> workd(3 * N, 0.0);
   std::vector<Real> workl(lworkl, 0.0);
-  std::vector<Real> d((nev + 1));
-  std::vector<Real> z((N + 1) * (nev + 1));
+  std::vector<Real> d(nev);
+  std::vector<Real> z(ldz * nev);
+  std::vector<a_int> select(ncv); // since HOWMNY = 'A', only used as workspace here
 
-  std::array<a_int, 11> iparam{};
+  a_int iparam[11], ipntr[11];
 
-  iparam[0] = 1;
-  iparam[2] = 10 * N;
-  iparam[3] = 1;
-  iparam[4] = 0;  // number of ev found by arpack.
```

Comment:
Can you let iparam[4] initialized to 0?
Initialisation is always good thing even if it's an output.